src/services_models/reserves.jl

```diff
@@ -5,57 +5,75 @@ struct RangeReserve <: AbstractReservesFormulation end
 This function add the variables for reserves to the model
 """
 function activeservice_variables!(psi_container::PSIContainer,
-                                  service::SR,
-                                  devices::Vector{<:PSY.Device}) where SR<:PSY.Reserve
+                                  services::IS.FlattenIteratorWrapper{SR},
+                                  service_devices::Dict{NamedTuple{(:type, :name),Tuple{DataType,String}},PSY.ServiceContributingDevices}) where SR<:PSY.Reserve
```

Comment:
Why are you creating the variables for all the services here?

Reply:
I'm only creating the variables for all services of a particular type. Mostly to be consistent with all of the device construction.